tests/cpp/test_multidevice_transformer.cpp

```diff
@@ -126,134 +132,305 @@ void validate(
 }
 } // namespace
 
-TEST_P(DistributedTransformerTest, MLP_Layer) {
-  auto [use_aten_matmul, dtype] = GetParam();
-  std::unique_ptr<Fusion> fusion = std::make_unique<Fusion>();
-  FusionGuard fg(fusion.get());
-  auto mesh = DeviceMesh::createForNumDevices(communicator_->size());
-  at::ScalarType at_dtype = data_type_to_aten(dtype);
+std::vector<at::Tensor> reference_mlp(
```

Comment:
Put this (and several other methods) in an anonymous namespace? Otherwise, they have external linkage, which unnecessarily slows down linking and can run into conflicts.